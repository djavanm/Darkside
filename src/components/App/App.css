@font-face {font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS") format("svg");
}

* {
  margin: 0;
}

main {
  display: grid;
  grid-template-columns: 1fr 4fr;
	grid-template-rows: 1fr 5fr;
  grid-template-areas:
    "nav header"
    "nav output";
  height: 100vh;
  background-image: url('http://eskipaper.com/images/star-background-1.jpg');
}

.App-header {
  grid-area: header;
  display: flex;
  justify-content: center;
  align-items: center;
}

h1 {
  font-size: 85px;
  font-family: 'STARWARS';
  color: yellow;
}

h1:hover {
  transform: scale(1.2) perspective(100px);
  transition: all 0.5s ease;
}
