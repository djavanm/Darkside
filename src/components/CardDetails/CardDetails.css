.CardDetails-div {
  background-color: black;
  width: 80%;
  height: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-radius: 10px;
  position: relative;
  opacity: 0.7;
}

.CardDetails_h2 {
  border-bottom: 3px solid yellow;
  padding: 5px;
  margin-bottom: 10px;
  color: yellow;
  font-size: 36px;
}

.CardDetails-div p {
  font-size: 24px;
  margin-bottom: 5px;
  color: yellow;
}

.CardDetails_li {
  margin-bottom: 5px;
  color: yellow;
  font-size: 18px;
}

.favorite {
  background: yellow;
}

.back-btn {
  text-decoration: none;
  color: black;
  font-size: 24px;
  position: absolute;
  top: 10px;
  left: 10px;
  color: yellow;
}

.back-btn:hover {
  color: darkblue;
}