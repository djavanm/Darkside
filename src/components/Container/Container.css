output {
  grid-area: output;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
  padding: 5px 20px;
}

.p_add-favorite {
  font-size: 42px;
  font-family: 'STARWARS';
  color: #FFF;
}

.link {
  text-decoration: none;
  color: black;
}


