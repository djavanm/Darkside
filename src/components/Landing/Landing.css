.Landing_crawl {
  width: 100%;
  background-color: lightblue;
}

.Landing_h2 {
  font-size: 120px;
  margin-bottom: 50px;
  text-align: center;
  text-transform: uppercase;
}

.Landing_p {
  font-size: 60px;
}

.crawler {
  background: transparent !important;
}

.crawler div {
  background: transparent !important;
  min-height: 40vh !important;
}