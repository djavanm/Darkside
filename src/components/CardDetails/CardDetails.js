import React from 'react';
import './CardDetails.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CardDetails = ({ name, homeworld, terrain, climate, species, homePopulation, type, residents, population, language, model, vehicleClass, passengers, favorites, theme, rotationPeriod, orbitalPeriod, diameter, gravity, hairColor, eyeColor, birthYear, cost, maxSpeed, cargoCapacity }) => {
  const isDark = theme === 'dark' ? 'dark-card' : '';
  let isFavorite = favorites.map(favorite => favorite.name).includes(name) ? 'favorite' : ''; 
  isFavorite = theme === 'dark' && isFavorite ? 'dark-favorite' : isFavorite; 
  let residentNames = undefined;
  if (residents) {
    residentNames = residents.map((resident, index) => {
      return <li key={index} className="CardDetails_li">{resident}</li>
    })
  }
  return (
    <div className={`CardDetails-div ${isFavorite} ${isDark}`}>
      <Link to={`/${type}`} className='back-btn'>â—€ back</Link>
      <h2 className="CardDetails_h2">{name}</h2>
      {!name && <p>404: No data found at this location!</p>}
      {homeworld && <p>Homeworld: {homeworld}</p>}
      {terrain && <p>Terrain: {terrain}</p>}
      {climate && <p>Climate: {climate}</p>}
      {homePopulation && <p>Population: {homePopulation}</p>}
      {species && <p>Species: {species}</p>}
      {population && <p>Population: {population}</p>}
      {language && <p>Language: {language}</p>}
      {model && <p>Model: {model}</p>}
      {vehicleClass && <p>Class: {vehicleClass}</p>}
      {passengers && <p>Number of Passengers: {passengers}</p>}
      {residents && !!residents.length && <div><p>Residents: </p><ul>{residentNames}</ul></div>}
      {orbitalPeriod && <p>Orbital Period: {orbitalPeriod}</p>}
      {rotationPeriod && <p>Rotation Period: {rotationPeriod}</p>}
      {diameter && <p>Diameter: {diameter}</p>}
      {gravity && <p>Gravity: {gravity}</p>}
      { hairColor && <p>Hair Color: {hairColor}</p> }
      { eyeColor && <p>Eye Color: {eyeColor}</p> }
      { birthYear && <p>Birth: {birthYear}</p> }
      { cost && <p>Cost: {cost} Credits</p> }
      { maxSpeed && <p>Max Atmospheric Speed: {maxSpeed} km/h</p> }
      { cargoCapacity && <p>Cargo Capacity: {cargoCapacity} tons</p> }
    </div>
  )
}

CardDetails.propTypes = {
  name: PropTypes.string,
  homeworld: PropTypes.string,
  terrain: PropTypes.string,
  climate: PropTypes.string,
  species: PropTypes.string,
  homePopulation: PropTypes.string,
  type: PropTypes.string,
  residents: PropTypes.array,
  population: PropTypes.string,
  language: PropTypes.string,
  model: PropTypes.string,
  vehicleClass: PropTypes.string,
  passengers: PropTypes.string,
  favorites: PropTypes.array,
  theme: PropTypes.string, 
  rotationPeriod: PropTypes.string, 
  orbitalPeriod: PropTypes.string,
  diameter: PropTypes.number,
  gravity: PropTypes.string,
  hairColor: PropTypes.string, 
  eyeColor: PropTypes.string, 
  birthYear: PropTypes.string,
  cost: PropTypes.string,
  maxSpeed: PropTypes.string, 
  cargoCapacity: PropTypes.string
}
export default CardDetails;
