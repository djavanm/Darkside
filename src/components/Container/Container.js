import React from 'react';
import Card from '../Card/Card'
import './Container.css';

const Container = ({ data, type, toggleFavorite, favorites }) => {
  const dataCards = data.map((card, index) => {
     return <Card className='Card-section 'data={card} type={type} key={index} toggleFavorite={toggleFavorite} favorites={favorites} />
  })
  return (
    <output>
      {type === 'favorites' && !favorites.length && <p className="p_add-favorite">Click a card to favorite</p>}
      {dataCards}
    </output>
  )
}

export default Container;