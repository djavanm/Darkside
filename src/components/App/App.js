import React, { Component } from 'react';
import Nav from '../Nav/Nav';
import Container from '../Container/Container';
import Landing from '../Landing/Landing';
import CardDetails from '../CardDetails/CardDetails';
import apiCalls from '../Util/apiCalls';
import { Route, Link } from 'react-router-dom';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      planets: [],
      people: [],
      vehicles: [],
      favorites: [],
      crawl: {},
      isLoading: true
    }
  }

  toggleFavorite = (card) => {
    const { favorites } = this.state;
    favorites.map(favorite => favorite.name).includes(card.name) ? this.removeFavorite(card) : this.addFavorite(card);
  }

  addFavorite = (card) => {
    const { favorites } = this.state;
    this.setState({ favorites: [...favorites, card]});
  }

  removeFavorite = (card) => {
    const { favorites } = this.state;
    const filteredFavorites = favorites.filter(favorite => favorite.name !== card.name)
    this.setState({ favorites: filteredFavorites});

  }

  componentDidMount() {
    let randomNumber = Math.floor((Math.random() * 7) + 1)
    fetch(`https://swapi.co/api/films/${randomNumber}`)
      .then(res => res.json())
      .then(data => this.setState({ crawl: { title: data.title, body: data.opening_crawl } }))
  

    fetch('https://swapi.co/api/planets/')
      .then(res => res.json())
      .then(data => apiCalls.cleanPlanetData(data.results))
      .then(planets => this.setState({ planets, isLoading: false }))

    fetch('https://swapi.co/api/people/')
      .then(res => res.json())
      .then(data => apiCalls.cleanPeopleData(data.results))
      .then(data => apiCalls.getPeopleHomes(data))
      .then(data => apiCalls.getSpecies(data))
      .then(people => this.setState({ people }))

    fetch('https://swapi.co/api/vehicles/')
      .then(res => res.json())
      .then(data => apiCalls.cleanVehicles(data.results))
      .then(vehicles => this.setState({ vehicles }))
  
    }
    
    render() {
      console.log(this.state)
      const { planets, people, vehicles, favorites, crawl, isLoading } = this.state;
      return (
        <main className="App-main">
        <header className="App-header">
          <Link to='/' className="link">
            <h1>LightSide</h1>
          </Link> 
        </header>
        <Nav favorites={favorites}/>
        <Route exact path='/' render={() => <Landing film={crawl}/> }/>
        <Route exact path='/planets' render={() => <Container data={planets} type="planets" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/people' render={() => <Container data={people} type="people" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/vehicles' render={() => <Container data={vehicles} type="vehicles" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/favorites' render={() => <Container data={favorites} type="favorites" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route path='/planets/:name' render={({ match }) => {
            let targetPlanet = this.state.planets.find(planet => planet.name.replace(/\s/g, '') == match.params.name);
            return <CardDetails {...targetPlanet} type="planets" />
        }} />
        <Route path='/people/:name' render={({ match }) => {
            let targetPerson = this.state.people.find(person => person.name.replace(/\s/g, '') == match.params.name);
            return <CardDetails {...targetPerson} type="people" favorites={favorites}/>
        }} />
        <Route path='/vehicles/:name' render={({ match }) => {
            let targetVehicle = this.state.vehicles.find(vehicle => vehicle.name.replace(/\s/g, '').replace("/", "") == match.params.name);
            return <CardDetails {...targetVehicle} type="vehicles" favorites={favorites}/>
        }} />
        <Route path='/favorites/:name' render={({ match }) => {
            let targetFavorite = this.state.favorites.find(favorite => favorite.name.replace(/\s/g, '').replace("/", "") == match.params.name);
            return <CardDetails {...targetFavorite} type="favorites" favorites={favorites}/>
        }} />
      </main>
    );
  }
}

export default App;
