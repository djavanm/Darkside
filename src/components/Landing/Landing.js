import React from 'react';
import Crawl from 'react-star-wars-crawl';
import 'react-star-wars-crawl/lib/index.css'
import './Landing.css';
import PropTypes from 'prop-types';

const Landing = ({ film }) => {
  return (
    <Crawl className="Landing_crawl">
      <h2 className="Landing_h2">{film.title}</h2>
      <p className="Landing_p">{film.body}</p>
    </Crawl>
  );
}

Landing.propTypes = {
  film: PropTypes.object.isRequired,
}
export default Landing;