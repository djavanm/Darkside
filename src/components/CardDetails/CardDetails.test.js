import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App/App';
import CardDetails from '../CardDetails/CardDetails'
import Landing from '../Landing/Landing';
import Container from '../Container/Container';
import { MemoryRouter } from 'react-router'
import { Route } from 'react-router-dom';

describe('Card Details', () => {
  it('should show Card Details component for a specific path', () => {
    const component = mount(<MemoryRouter initialEntries={['/people/LeiaOrgana']} >
      <App />
    </MemoryRouter>
    );
    expect(component.find(CardDetails)).toHaveLength(1);
  });
});

