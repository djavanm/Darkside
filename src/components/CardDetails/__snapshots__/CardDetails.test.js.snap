// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Details should match the snapshot when a person is displayed with all data passed in correctly 1`] = `
<div
  className="CardDetails-div  "
>
  <Link
    className="back-btn"
    to="/people"
  >
    ◀ back
  </Link>
  <h2
    className="CardDetails_h2"
  >
    Luke Skywalker
  </h2>
  <p>
    Homeworld: 
    Tatooine
  </p>
  <p>
    Population: 
    200000
  </p>
  <p>
    Species: 
    Human
  </p>
  <p>
    Language: 
    Galactic Basic
  </p>
</div>
`;

exports[`Card Details should match the snapshot when a planet is displayed with all data passed in correctly 1`] = `
<div
  className="CardDetails-div  "
>
  <Link
    className="back-btn"
    to="/planets"
  >
    ◀ back
  </Link>
  <h2
    className="CardDetails_h2"
  >
    Alderaan
  </h2>
  <p>
    Terrain: 
    grasslands, mountains
  </p>
  <p>
    Climate: 
    temperate
  </p>
  <p>
    Population: 
    2000000000
  </p>
  <div>
    <p>
      Residents: 
    </p>
    <ul>
      <li
        className="CardDetails_li"
        key="0"
      >
        Leia Organa
      </li>
      <li
        className="CardDetails_li"
        key="1"
      >
        Bail Prestor Organa
      </li>
      <li
        className="CardDetails_li"
        key="2"
      >
        Raymus Antilles
      </li>
    </ul>
  </div>
</div>
`;

exports[`Card Details should match the snapshot when a vehicle is displayed with all data passed in correctly 1`] = `
<div
  className="CardDetails-div  "
>
  <Link
    className="back-btn"
    to="/vehicles"
  >
    ◀ back
  </Link>
  <h2
    className="CardDetails_h2"
  >
    TIE bomber
  </h2>
  <p>
    Model: 
    IE/sa bomber
  </p>
  <p>
    Class: 
    space/planetary bomber
  </p>
  <p>
    Number of Passengers: 
    0
  </p>
</div>
`;
